src/main/java/org/nuxeo/ecm/csv/CSVImporterImpl.java

```diff
@@ -18,21 +18,67 @@
 package org.nuxeo.ecm.csv;
 
 import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
 
+import org.nuxeo.ecm.core.api.ClientException;
 import org.nuxeo.ecm.core.api.CoreSession;
 import org.nuxeo.ecm.core.work.api.Work;
 import org.nuxeo.ecm.core.work.api.Work.State;
 import org.nuxeo.ecm.core.work.api.WorkManager;
+import org.nuxeo.ecm.csv.descriptor.AcceptedTypeDescriptor;
+import org.nuxeo.ecm.csv.descriptor.CSVFileHandlerDescriptor;
+import org.nuxeo.ecm.csv.descriptor.CSVFileHandlerDescriptorComparator;
 import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.model.ComponentInstance;
+import org.nuxeo.runtime.model.DefaultComponent;
+import org.richfaces.model.UploadedFile;
 
 /**
  * @since 5.7
  */
-public class CSVImporterImpl implements CSVImporter {
+public class CSVImporterImpl extends DefaultComponent implements CSVImporter {
+    private static final String FILE_HANDLERS_EP = "fileHandlers";
+
+    private static final String ACCEPTED_TYPES_EP = "acceptedTypes";
+
+    private final Map<String, CSVFileHandlerDescriptor> fileHandlers = new HashMap<>();
+
+    private final Set<String> acceptedTypes = new HashSet<>();
+
+    public CSVImporterImpl() {
+        super();
+    }
+
+    @Override
+    public void registerContribution(Object contribution, String extensionPoint, ComponentInstance contributor) {
+        if (FILE_HANDLERS_EP.equals(extensionPoint) && contribution instanceof CSVFileHandlerDescriptor) {
+            CSVFileHandlerDescriptor descriptor = (CSVFileHandlerDescriptor) contribution;
+            String id = descriptor.getId();
+            if (descriptor.getEnabled()) {
+                fileHandlers.put(id, descriptor);
+            } else {
+                fileHandlers.remove(id);
+            }
```

Comment:
This kind of registration / unregistration code is still quite common in the platform but does not adequately take into account unregistering things in different order than the registration (partial undeploy of a bundle). It's better to use SimpleContributionRegistry (see the many subclasses that use it). I'll accept the pull request without this fix though.